src/Database/Log/QueryLogger.php

```diff
@@ -68,7 +68,19 @@ protected function _interpolate($query)
                 return $p ? '1' : '0';
             }
 
-            return is_string($p) ? "'$p'" : $p;
+            if (is_string($p)) {
+                $replacements = [
+                    '$' => '\\$',
+                    '\\' => '\\\\\\\\',
+                    "'" => "''",
+                ];
+
+                $p = strtr($p, $replacements);
```

Comment:
Instead of manually replacing specific characters it would be better to instead use `preg_quote()` when generating the `$keys` array below.